### PR TITLE
Add New Grants endpoints

### PR DESCRIPTION
**Pull request by Norio4**
### Changes

- Adds new Grants endpoints

### References

- https://auth0.com/docs/api/management/v2#!/Grants/get_grants

### CODE REVIEW

**Comment by Norio4** [comment]
Dear developers of a great library.

I'm sorry if I'm not clear enough about this pull request.
I have created a pull request based on PR #293.

There is a Grants API in the Auth0 Management API Doc, but it was not implemented in this library. So I have added a module to call the Grants API.

Thanks!

**Comment by davidpatrick** [comment]
Thank you @Norio4 for raising this :bow:  I will review this week

File: lib/auth0/api/v2/grants.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        # Retrieve the grants associated with your account.
+        # @see https://auth0.com/docs/api/management/v2#!/Grants/get_grants
+        # @param client_id [string] The client_id of the grants to retrieve.
+        # @param user_id [string]  The user_id of the grants to retrieve.
+        # @param audience [string] The audience of the grants to retrieve.
+        # @param page [int] The page index of the results to return. First page is 0.
+        # @param per_page [int] The number of results per page. Paging is disabled if parameter not sent.
+        # @param include_totals [boolean] Return results inside an object that contains the total result count (true) or as a direct array of results (false, default).
+        # @return [json] Returns the grants.
+        def grants(client_id: nil, user_id: nil, audience: nil, page: nil, per_page: nil, include_totals: false)
```
Comment (davidpatrick) [inline lib/auth0/api/v2/grants.rb:16]:
```suggestion
        def grants(client_id: nil, user_id: nil, audience: nil, page: nil, per_page: nil, include_totals: nil)
```

File: spec/lib/auth0/api/v2/grants_spec.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    it 'is expected to get /api/v2/grants/' do
+      expect(@instance).to receive(:get).with(
+        '/api/v2/grants',
+        client_id: nil,
+        user_id: nil,
+        audience: nil,
+        page: nil,
+        per_page: nil,
+        include_totals: false
```
Comment (davidpatrick) [inline spec/lib/auth0/api/v2/grants_spec.rb:21]:
```suggestion
        include_totals: nil
```

File: spec/lib/auth0/api/v2/grants_spec.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      audience = "https://samples.auth0.com/api/v2/"
+
+      expect(@instance).to receive(:get).with(
+        '/api/v2/grants',
+        client_id: '1',
+        user_id: nil,
+        audience: audience,
+        page: nil,
+        per_page: nil,
+        include_totals: false
```
Comment (davidpatrick) [inline spec/lib/auth0/api/v2/grants_spec.rb:36]:
```suggestion
        include_totals: nil
```

File: spec/lib/auth0/api/v2/grants_spec.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    it 'is expected to send get /api/v2/grants/ with client_id and user_id' do
+      expect(@instance).to receive(:get).with(
+        '/api/v2/grants',
+        client_id: '1',
+        user_id: '1',
+        audience: nil,
+        page: nil,
+        per_page: nil,
+        include_totals: false
```
Comment (davidpatrick) [inline spec/lib/auth0/api/v2/grants_spec.rb:49]:
```suggestion
        include_totals: nil
```

File: spec/lib/auth0/api/v2/grants_spec.rb [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    it 'is expected to send get /api/v2/grants/ with pagination' do
+      expect(@instance).to receive(:get).with(
+        '/api/v2/grants',
+        client_id: nil,
+        user_id: nil,
+        audience: nil,
+        page: 1,
+        per_page: 2,
+        include_totals: false
```
Comment (davidpatrick) [inline spec/lib/auth0/api/v2/grants_spec.rb:62]:
```suggestion
        include_totals: nil
```

**Review by davidpatrick — CHANGES_REQUESTED** [review]
Great work here.  Just small change to include_totals

**Review by davidpatrick — APPROVED** [review]

**Pull request merged**